titus-server-runtime/src/main/java/com/netflix/titus/runtime/jobactivity/JobActivityPublisherRecordUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.runtime.jobactivity;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.netflix.titus.api.jobactivity.store.JobActivityPublisherRecord;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.runtime.endpoint.common.LogStorageInfo;
+import com.netflix.titus.runtime.endpoint.v3.grpc.V3GrpcModelConverters;
+
+public class JobActivityPublisherRecordUtils {
```

Comment:
This should be in the store implementation.

Reply:
This is currently being used the the jOOQ store implementation in `titus-ext` and the in-memory store in `titus-server-master`. Unless I move to the store interface in `titus-api`, I'm not sure how to move it into the store implementation without duplicating code. These are effectively helper methods for the `JobActivityPublisherRecord` data type but seems like we're trying to keep implementation logic out of the data classes. Open to suggestions.